src/basic/mkdir.c

```diff
@@ -204,10 +205,13 @@ int mkdir_p_safe(const char *prefix, const char *path, mode_t mode, uid_t uid, g
         return mkdir_p_internal(prefix, path, mode, uid, gid, flags, mkdirat_errno_wrapper);
 }
 
-int mkdir_p_root(const char *root, const char *p, uid_t uid, gid_t gid, mode_t m, char **subvolumes) {
+int mkdir_p_root_full(const char *root, const char *p, uid_t uid, gid_t gid, mode_t m, usec_t ts, char **subvolumes) {
         _cleanup_free_ char *pp = NULL, *bn = NULL;
         _cleanup_close_ int dfd = -EBADF;
+        _cleanup_close_ int nfd = -EBADF;
         int r;
+        struct timespec dtime[2];
+        struct timespec nftime[2];
```

Comment:
Please reduce the scope of these vars, i.e.  move these into the corresponding if blocks